### ISSUE TITLE
L007 refers to deprecated `operator_new_lines`

### ISSUE THREAD

**Issue by tvoipio** [sqlfluff/sqlfluff#4166]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Links or command line

https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.rules.Rule_L007

### Issue/Suggested Improvement

Documentation for rule L007 (operators before/after newlines) consistently refers to configuration key `operator_new_lines`. Setting this configuration key in `[sqlfluff:rules:L007]` results in

```
WARNING: Config file $PWD/.sqlfluff set a deprecated config value `rules:L007:operator_new_lines`. This will be removed in a later release. This has been mapped to `layout:type:binary_operator:line_position` set to a value of `leading` for this run. Please update your configuration to remove this warning.

Use the line_position config in the appropriate sqlfluff:layout section (e.g. sqlfluff:layout:type:binary_operator).

See https://docs.sqlfluff.com/en/stable/configuration.html for more details.
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Fixed by #5314
